src/components/sensors/Sensors.js

```diff
@@ -5,32 +5,143 @@ import SensorForm from './sensor/SensorForm.js';
+import FormControl from '@material-ui/core/FormControl';
+import Input from '@material-ui/core/Input';
+import MenuItem from '@material-ui/core/MenuItem';
+import Grid from '@material-ui/core/Grid';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Collapse from '@material-ui/core/Collapse';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import FilterList from '@material-ui/icons/FilterList';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  formControl: {
+    margin: theme.spacing.unit,
+    minWidth: 120,
+  },
+  selectEmpty: {
+    marginTop: theme.spacing.unit * 2,
+  },
+});
 
 class Sensors extends Component {
   constructor(props) {
     super(props);
+    this.filterQuery = `owner==${props.user.username};`,
     this.state = {
+      open: false,
       modalAddSensor: false,
       isCardsView: true,
+      filter: {
+        domain: 'all',
+        owner: props.user.username,
+        visibility: 'all',
+        status: 'all'
+      }
     };
+
   }
 
+  handleClick = () => {
+    this.setState(state => ({ open: !state.open }));
+  };
+
   componentWillMount() {
-    this.props.getSensors({ limit: 1000 });
+    this.props.getSensors({ q: this.filterQuery, limit: 1000 });
+    this.props.getSensorAttributes({ limit: 1000 });
     this.interval = setInterval(() => {
-      this.props.getSensors({ limit: 1000 });
+      this.props.getSensors({ q: this.filterQuery, limit: 1000 });
     }, config.delayRefresh);
   }
 
   componentWillUnmount() {
     clearInterval(this.interval);
+
+  }
+
+  compare(a, b) {
+    if (a < b)
+      return -1;
+    if (a > b)
+      return 1;
+    return 0;
+  }
+
+  handleFilter = (field, event) => {
+
+    if (!this.filterQuery) {
+      clearInterval(this.interval);
+      this.interval = setInterval(() => {
+        this.props.getSensors({ limit: 1000, q: this.filterQuery });
+      }, config.delayRefresh);
+    }
+
+    const value = event.target.value;
+    var filter = this.state.filter;
+    let statusFilter;
+    switch (field) {
+      case "domain":
+        {
+          filter.domain = value;
+          break;
+        }
+      case "owner":
+        {
+          filter.owner = value;
+          break;
+        }
+      case "visibility":
+        {
+          filter.visibility = value;
+          break;
+        }
+      case "status":
+        {
+          filter.status = value;
+          var yesterday = new Date(Date.now() - 86400 * 1000).toISOString();
+          if (value == 'all') {
+            statusFilter = `dateModified>2000-01-01T00:00:00.00Z`
+          }
+          else if (value == 'new') {
+            statusFilter = `dateCreated>${yesterday}`
```

Comment:
delay should be taken from config file (delaySensorNodeNew, delaySensorNodeInactive)